### ISSUE TITLE
PHP 8.1: Deprecation notice due to passing null to internal explode function

### ISSUE THREAD

**Issue by Sdfendor** [Shardj/zf1-future#181]
First of all I'm unclear about the current PHP 8.1 situation with this project. Currently only PHP 8.0 is supported but I thought I report issues with the recently released 8.1 nonetheless. If this thought was wrong I'm sorry. 

PHP 8.1 introduced this deprecation for all internal functions: [Passing null to non-nullable internal function parameters is deprecated](https://php.watch/versions/8.1/internal-func-non-nullable-null-deprecation)

The method `getExtension` in `Zend_Validate_File_Extension`:
```php
/**
 * Returns the set file extension
 *
 * @return array
 */
public function getExtension()
{
    return explode(',', $this->_extension);
}
```
causes such a deprecation compulsory if used (`Deprecated: explode(): Passing null to parameter #2 ($string) of type string is deprecated in Zend/Validate/File/Extension.php on line 127`). 

Although the property is correctly initialized in line 55 with an empty string the problem lies with `setExtension` which nulls the member prior to calling `getExtension` and is called in the constructor of the class. 
```php
/**
 * Sets the file extensions
 *
 * @param  string|array $extension The extensions to validate
 * @return Zend_Validate_File_Extension Provides a fluent interface
 */
public function setExtension($extension)
{
    $this->_extension = null;
    $this->addExtension($extension);
    return $this;
}
```

**Comment by Shardj** [comment]
Thanks this is helpful, I'll be looking at an upgrade to PHP 8.1 in January

**Comment by Sdfendor** [comment]
Fixed in #182.
